src/expr/impl/src/scalar/jsonb_object.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use jsonbb::Builder;
+use risingwave_common::types::{JsonbVal, ListRef};
+use risingwave_common::util::iter_util::ZipEqFast;
+use risingwave_expr::{function, ExprError, Result};
+
+/// Builds a JSON object out of a text array.
+///
+/// The array must have either exactly one dimension with an even number of members,
+/// in which case they are taken as alternating key/value pairs, or two dimensions
+/// such that each inner array has exactly two elements, which are taken as a key/value pair.
+/// All values are converted to JSON strings.
+///
+/// # Examples
+///
+/// ```slt
+/// query T
+/// select jsonb_object('{a, 1, b, def, c, 3.5}' :: text[]);
+/// ----
+/// {"a": "1", "b": "def", "c": "3.5"}
+///
+/// query error array must have even number of elements
+/// select jsonb_object('{a, 1, b, "def", c}' :: text[]);
+/// ```
+#[function("jsonb_object(varchar[]) -> jsonb")]
+fn jsonb_object_1d(array: ListRef<'_>) -> Result<JsonbVal> {
+    if array.len() % 2 == 1 {
+        return Err(ExprError::InvalidParam {
+            name: "array",
+            reason: "array must have even number of elements".into(),
+        });
+    }
+    let mut builder = Builder::<Vec<u8>>::new();
+    builder.begin_object();
+    for value in array.iter() {
+        match value {
+            Some(s) => builder.add_string(s.into_utf8()),
+            None => builder.add_null(),
+        }
+    }
```

Comment:
Key (elements at even position) cannot be null in PostgreSQL:
```
test=# select jsonb_object(array[null, 'b']);
ERROR:  null value not allowed for object key

test=# select jsonb_object(array['a', null]);
 jsonb_object 
--------------
 {"a": null}
(1 row)
```

Reply:
Good catch, thanks!